Source/Core/Core/PowerPC/Jit64/Jit_FloatingPoint.cpp

```diff
@@ -56,6 +56,144 @@ void Jit64::SetFPRFIfNeeded(X64Reg xmm)
 		SetFPRF(xmm);
 }
 
+void Jit64::HandleNaNs(UGeckoInstruction inst, X64Reg xmm_out, X64Reg xmm)
+{
+	//                      | PowerPC  | x86
+	// ---------------------+----------+---------
+	// input NaN precedence | 1*3 + 2  | 1*2 + 3
+	// generated QNaN       | positive | negative
+	//
+	// Dragon Ball: Revenge of King Piccolo requires generated NaNs
+	// to be positive, so we'll have to handle them manually.
+
+	if (!SConfig::GetInstance().m_LocalCoreStartupParameter.bAccurateNaNs)
+	{
+		if (xmm_out != xmm)
+			MOVAPD(xmm_out, R(xmm));
+		return;
+	}
+
+	_assert_(xmm == XMM1);
+
+	std::vector<u32> inputs;
+	u32 a = inst.FA, b = inst.FB, c = inst.FC;
+	for (u32 i : {a, b, c})
+	{
+		if (!js.op->fregsIn[i])
+			continue;
+		if (std::find(inputs.begin(), inputs.end(), i) == inputs.end())
+			inputs.push_back(i);
+	}
+	if (inst.OPCD != 4)
+	{
+		// not paired-single
+		UCOMISD(xmm, R(xmm));
+		FixupBranch handle_nan = J_CC(CC_P, true);
+		SwitchToFarCode();
+			SetJumpTarget(handle_nan);
+			std::vector<FixupBranch> fixups;
+			for (u32 x : inputs)
+			{
+				MOVDDUP(xmm, fpr.R(x));
+				UCOMISD(xmm, R(xmm));
+				fixups.push_back(J_CC(CC_P));
+			}
+			MOVDDUP(xmm, M(psGeneratedQNaN));
+			for (FixupBranch fixup : fixups)
+				SetJumpTarget(fixup);
+			FixupBranch done = J(true);
+		SwitchToNearCode();
+		SetJumpTarget(done);
+	}
+	else
+	{
+		// paired-single
+		X64Reg mask_in = fpr.GetFreeXReg();
+		fpr.FlushLockX(mask_in);
+		X64Reg tmp1 = fpr.GetFreeXReg();
+		fpr.FlushLockX(tmp1);
+		X64Reg tmp2 = fpr.GetFreeXReg();
+		fpr.FlushLockX(tmp2);
+		avx_op(&XEmitter::VCMPPD, &XEmitter::CMPPD, mask_in, R(xmm), R(xmm), CMP_UNORD);
+		if (cpu_info.bSSE4_1 && cpu_info.bAVX)
```

Comment:
Yes, but this way it is obvious that the code uses both, in case anyone feels like touching this ever again.